Task_1/task1_file_parser.py

```diff
@@ -0,0 +1,49 @@
+#!/bin/python3
+import os
+import sys
+from datetime import datetime
+
+#file_path = sys.argv[1]
+file_path = 'C:\PythonMentorship\Task_1\input\log_data\log_data.txt'
+
+def group_by_date_and_IP (record_list):
+    grouped_by_date = {}
+    for record in record_list:
+        if record["Datetime"].date() not in grouped_by_date:
+            grouped_by_date[record["Datetime"].date()] = {}
+        grouped_by_IP = grouped_by_date[record["Datetime"].date()]
+        if record["IP"] not in grouped_by_IP:
+            grouped_by_IP[record["IP"]] = record["Size"]
+        else:
+            grouped_by_IP[record["IP"]] += record["Size"]
+    return grouped_by_date
+
+
+def print_top_downloader(grouped_by_date):
+    print('IP address of the top downloader: ')
+    for date, ip in grouped_by_date.items():
+        top_downloader = max(ip, key = ip.get)
+        print(f'{date} - {top_downloader}')
+
+
+def get_least_busy_hours(record_list):
+    list_of_hours = dict.fromkeys(range(00,24),0)

```

Comment:
Please note that all numbers prepended with a "0" are treated as hexadecimal (in particular, because of that 09 is an invalid number).
Please remove the leading zero to avoid potential issues.

Reply:
Done